File: c/bmi.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

float bmi(float height, float weight) {
  height /=100;
  return weight / (height * height);
}

void check(float value) {
  if (!(int)roundf(value))
  {
    printf("Should be a valid number\n");
    exit(1);
  }
}

char* judge(float bmi) {
  if (bmi >= 25)
    return "overweight";
  else if (bmi < 18.5)
    return "underweight";
  else
    return "normal";
}

int main() {
  float height, weight;

  printf("Enter weight: ");
  scanf("%f", &weight);
  check(weight);

  printf("Enter height: ");
  scanf("%f", &height);
  check(height);

  printf("%s\n", judge(bmi(height, weight)));

  return 0;
}

File: js/bmi.js
const readline = require('readline').createInterface({
  input: process.stdin,
  output: process.stdout
})

function enterWeight() {
  return new Promise((resolve) => {
    readline.question('Enter weight: ', (weight) => { resolve(parseFloat(weight)) })
  })
}

function enterHeight(weight) {
  return new Promise((resolve) => {
    readline.question('Enter height: ', (height) => { resolve([weight, parseFloat(height)]) })
  })
}

function printJudgement([weight, height]) {
  console.log(judge(bmi(height, weight)))
  readline.close()
}

function judge(bmi) {
  if (bmi >= 25)
    return "overweight"
  else if (bmi < 18.5)
    return "underweight"
  else
    return "normal"
}

function bmi(height, weight) {
  height /=100
  return weight / (height * height)
}

enterWeight().then(enterHeight).then(printJudgement)
